src/bionics.cpp

```diff
@@ -14,38 +14,51 @@
 #include <algorithm> //std::min
 #include <sstream>
 
-#define BATTERY_AMOUNT 100 // How much batteries increase your power
+namespace {
+constexpr int BATTERY_AMOUNT = 100; // How much batteries increase your power
+std::map<std::string, bionic_data> bionics;
+std::vector<std::string> faulty_bionics;
+} //namespace
 
-std::map<bionic_id, bionic_data *> bionics;
-std::vector<bionic_id> faulty_bionics;
-std::vector<bionic_id> power_source_bionics;
-std::vector<bionic_id> unpowered_bionics;
+bool is_valid_bionic(std::string const& id)
+{
+    return !!bionics.count(id);
+}
+
+bionic_data const& bionic_info(std::string const &id)
+{
+    auto const it = bionics.find(id);
+    if (it != bionics.end()) {
+        return it->second;
+    }
+
+    debugmsg("bad bionic id");
+
+    static bionic_data const null_value {"bad bionic", false, false, 0, 0, 0, 0, "bad_bionic", false};
+    return null_value;
+}
 
 void bionics_install_failure(player *u, int difficulty, int success);
 
 bionic_data::bionic_data(std::string nname, bool ps, bool tog, int pac, int pad, int pot,
-                          int ct, std::string desc, bool fault) : description(desc)
+                         int ct, std::string desc, bool fault
+) : name(std::move(nname)), description(std::move(desc)), power_activate(pac),
+    power_deactivate(pad), power_over_time(pot), charge_time(ct), faulty(fault),
+    power_source(ps), activated(tog || pac || ct), toggled(tog)
 {
-    name = nname;
-    power_source = ps;
-    activated = tog || pac || ct;
-    toggled = tog;
-    power_activate = pac;
-    power_deactivate = pad;
-    power_over_time = pot;
-    charge_time = ct;
-    faulty = fault;
 }
 
-bionic_id game::random_good_bionic() const
-{
-    std::map<std::string, bionic_data *>::const_iterator random_bionic;
-    do {
-        random_bionic = bionics.begin();
-        std::advance(random_bionic, rng(0, bionics.size() - 1));
-    } while (random_bionic->second->faulty);
-    return random_bionic->first;
-}
+// TODO this is unused and the existing implementation will loop forever if there are no
```

Comment:
"Unused code should be removed, not commented out. The version control system should be used in case anybody wants to get the old code back." Quote from some famous, I forgot who, but I totally agree.